proto/sifnode/clp/v1/tx.proto

```diff
@@ -219,11 +226,48 @@ message MsgSetSymmetryThresholdResponse {}
 message MsgCancelUnlock {
   string signer = 1;
   sifnode.clp.v1.Asset external_asset = 2
-  [ (gogoproto.moretags) = "yaml:\"external_asset\"" ];
+      [ (gogoproto.moretags) = "yaml:\"external_asset\"" ];
   string units = 3 [
     (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Uint",
     (gogoproto.nullable) = false
   ];
 }
 
-message MsgCancelUnlockResponse {}
\ No newline at end of file
+message MsgCancelUnlockResponse {}
+
+message MsgAddSwapAssetPermission {
+  string signer = 1;
+  sifnode.clp.v1.Asset asset = 2;
+  SwapPermission swap_permission = 3;
+}
+
+message MsgAddSwapAssetPermissionResponse {}
+
+message MsgRemoveSwapAssetPermission {
+  string signer = 1;
+  sifnode.clp.v1.Asset asset = 2;
+  SwapPermission swap_permission = 3;
+}
+
+message MsgRemoveSwapAssetPermissionResponse {}
+
+message MsgModifyLiquidityProtectionRates {
+  string signer = 1 [ (gogoproto.moretags) = "yaml:\"signer\"" ];
+  string current_rowan_liquidity_threshold = 2 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+}
+
+message MsgModifyLiquidityProtectionRatesResponse {}
+
+message MsgUpdateLiquidityProtectionParams {
+  string signer = 1 [ (gogoproto.moretags) = "yaml:\"signer\"" ];
+  string max_rowan_liquidity_threshold = 2 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
```

Comment:
thanks for your comment @timlind I choose to use Dec as it will be valued in USD or any other asset defined in `max_rowan_liquidity_threshold_asset` are you thinking of a better way to approach that?